experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyField.ts

```diff
@@ -202,10 +202,14 @@ export abstract class LazyField<TKind extends FieldKindTypes, TTypes extends All
 		);
 	}
 
-	public [Symbol.iterator](): IterableIterator<TypedNodeUnion<TTypes>> {
+	public boxedIterator(): IterableIterator<TypedNodeUnion<TTypes>> {
 		return this.mapBoxed((x) => x)[Symbol.iterator]();
 	}
 
+	public [Symbol.iterator](): IterableIterator<UnboxNodeUnion<TTypes>> {
```

Comment:
if we want to share code between this and map, I'd rather have this one be lazy, and have may implemented as `Array.from(this, f)`.

Reply:
doing it that way I think will also error instead of silently get out of date if you do it while editing, and will be easier to modify to make it do the right thing while editing in the future.

Same for boxed